src/main/java/com/hms_networks/americas/sc/extensions/system/time/LocalTimeOffsetCalculator.java

```diff
@@ -130,4 +130,21 @@ private static long parseLocalTimeResult() throws ParseException {
 
     return diffInMilliseconds;
   }
+
+  /**
+   * Converts the local epoch milliseconds timestamp to UTC epoch milliseconds. Flexy EBD historical
+   * data response will return integer timestamps in local time when "Record Data in UTC" is not
+   * enabled, so this method can be used to convert the local time to UTC time.
+   *
+   * @param localEpochMillis the local epoch milliseconds timestamp
+   * @return epoch milliseconds timestamp relative to UTC
+   * @throws InterruptedException if interrupted while attempting script expression export block
+   *     descriptor call
+   * @throws ParseException if unable to parse script expression export block descriptor call result
+   *     (local time)
+   */
+  public static long convertLocalEpochMillisToUtc(long localEpochMillis)
+      throws InterruptedException, ParseException {
+    return localEpochMillis + calculateLocalTimeOffsetMilliseconds();
```

Comment:
Should this use `getLocalTimeOffsetMilliseconds()` instead?

In its current form, this method would force a local time offset calculation each time it is invoked. If it's being invoked for every historical data EBD response, that can be very expensive performance-wise, and it is unnecessary to update the time offset at such a fast rate (potentially many times a minute).

Reply:
This still seems like a costly operation to perform each time. The name/Javadocs/appearance of the method also does not convey that it is updating the local time offset calculation. Based on the method name and Javadocs, I would expect that this method is just simply converting local epoch millis to UTC.

I understand the intention here, but I wonder whether there is a solution that does not put even further pressure on the already constrained Flexy by adding more EBD calls.

Additionally, this is also circumventing the SCTimeUtils class process for injecting time zone configurations, as well as the newly created `TimeZoneManager` class you're adding. It might make sense to align with those classes to ensure that if there is updated time offset information being pulled, we're also accordingly updating the JVM TZ configuration, too.

Reply:
I think you are correct the TimeZoneManager class, if used, should maintain the offset. 
Right now this method is only called when Rapid Catch up finds  data and data point timestamps are local. 
I will make the change.